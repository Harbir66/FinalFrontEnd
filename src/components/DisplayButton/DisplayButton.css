.display-button{
    width: 19vw;
    margin: 2vh 3vw 0;
    padding: 0 1vw;
    color: #777777;
    border: none;
    height: 6vh;
    border-radius: 1vh;
    font-size: 2vh;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.selected{
    background-color: #5905CE;
    color: white;
}