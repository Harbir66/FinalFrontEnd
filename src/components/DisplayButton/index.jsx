import React from 'react';
import './DisplayButton.css';
import PropTypes from 'prop-types';

function DisplayButton({ text, count, selected, onClick }) {
  const className = selected ? 'display-button selected' : 'display-button';
  return (
    <button onClick={onClick} type="button" className={className}>
      <span>{text}</span>
      <span>{count}</span>
    </button>
  );
}

export default DisplayButton;

DisplayButton.propTypes = {
  text: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired,
  selected: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};
