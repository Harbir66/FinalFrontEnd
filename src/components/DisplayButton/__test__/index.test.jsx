import React from 'react';
import { render } from '@testing-library/react';
import DisplayButton from '..';

describe('DisplayButton', () => {
  it('should render correctly when selected ', () => {
    const { asFragment } = render(
      <DisplayButton text={test} count={10} selected onClick={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render correctly when not selected ', () => {
    const { asFragment } = render(
      <DisplayButton
        text={test}
        count={10}
        selected={false}
        onClick={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
