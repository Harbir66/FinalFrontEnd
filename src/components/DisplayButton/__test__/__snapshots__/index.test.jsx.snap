// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayButton should render correctly when not selected  1`] = `
<DocumentFragment>
  <button
    class="display-button"
    type="button"
  >
    <span />
    <span>
      10
    </span>
  </button>
</DocumentFragment>
`;

exports[`DisplayButton should render correctly when selected  1`] = `
<DocumentFragment>
  <button
    class="display-button selected"
    type="button"
  >
    <span />
    <span>
      10
    </span>
  </button>
</DocumentFragment>
`;
